core/services/workflows/engine.go

```diff
@@ -704,9 +739,14 @@ func (e *Engine) executeStep(ctx context.Context, msg stepRequest) (*values.Map,
 		return nil, nil, err
 	}
 
+	config, err := configForStep(ctx, e.registry, e.localNode, step)
```

Comment:
Potential ticket (for the future minor perf optimization) - do this out of band when registry sync detects changes in DON capability config.